app/controllers/pxe_controller/pxe_servers.rb

```diff
@@ -259,37 +185,17 @@ def pxe_wimg_form_field_changed
     end
   end
 
-  private #######################
-
-  def pxe_server_validate_fields
-    if @edit[:new][:name].blank?
-      add_flash(_("Name is required"), :error)
-    end
-    if @edit[:new][:uri_prefix].blank?
-      add_flash(_("Depot Type is required"), :error)
-    end
-    if @edit[:new][:uri_prefix] == "nfs" && @edit[:new][:uri].blank?
-      add_flash(_("URI is required"), :error)
-    end
-    if @edit[:new][:uri_prefix] == "smb" || @edit[:new][:uri_prefix] == "ftp" ||
-       @edit[:new][:uri_prefix] == "s3"
-      pxe_server_validate_creds
+  def pxe_server_async_cred_validation
+    begin
+      PxeServer.verify_depot_settings(params[:pxe])
+    rescue => bang
+      render :json => {:status => 'error', :message => _("Error during 'Validate': %{error_message}") % {:error_message => bang.message}}
```

Comment:
What about
```ruby
render :json => { ... }, :status => :bad_request
```

then you don't need to check if `status === 'error'` or not, and you can use `.catch` block instead of that.

Also, you don't need `else` branch, you can write the successful render after the verify method.

Reply:
> What about
> 
> ```ruby
> render :json => { ... }, :status => :bad_request
> ```
> 
> then you don't need to check if `status === 'error'` or not, and you can use `.catch` block instead of that.
> 
> Also, you don't need `else` branch, you can write the successful render after the verify method.

I was actually discussing this with @himdel and he advice me to use 200 response with error status.